### PR TITLE
Update Mac to draft 05

### PR DESCRIPTION
**Pull request by anvox**
Remake https://github.com/oauth-xx/oauth2/pull/196 with a little change in detail. 
See comments. 

Implement https://github.com/oauth-xx/oauth2/issues/194

### CODE REVIEW

**Comment by anvox** [comment]
I see https://github.com/oauth-xx/oauth2/pull/196 inactive for a long time, so I remake.

File: lib/oauth2/mac_token.rb [hunk 1]
```diff
@@ -29,6 +29,9 @@ def self.from_access_token(token, secret, options = {})
     # @option opts [FixNum, String] :algorithm (hmac-sha-256) the algorithm to use for the HMAC digest (one of 'hmac-sha-256', 'hmac-sha-1')
     def initialize(client, token, secret, opts = {})
       @secret = secret
+      @seq_nr = SecureRandom.random_number(2 ** 64)
```
Comment (anvox) [inline lib/oauth2/mac_token.rb:32]: > Sequence numbers fall in the range 0 through 2^64 - 1 and wrap to zero following the value 2^64 - 1.

As [SecureRandom.random_number(n)](https://apidock.com/ruby/SecureRandom/random_number/class) generate number => 0 and < n, we needn'nt minus 1

**Comment by pboling** [comment]
@anvox This looks great!  This appears to be a breaking change?  Would you suggest a minor release, or a major release?  How "breaking" is this change?

**Comment by anvox** [comment]
@pboling I think it should be a major change. By this change, we no longer support MAC token draft < 3. Clients must change from using nonce to seq-nr.

**Comment by pboling** [comment]
@anvox ok, this will be a new major release of 2.0.0, and the old major milestones will all get bumped up by +1.0.0

**Pull request merged**